### PR TITLE
CI: remove win prerelease azure + add py39

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

Tweaks what jobs we run on azure.

### CODE REVIEW

**Comment by QuLogic** [comment]
Replaces #19353?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -11,21 +11,27 @@ strategy:
     Linux_py38:
       vmImage: 'ubuntu-16.04'
       python.version: '3.8'
+    Linux_py38:
```
Comment (timhoffm) [inline azure-pipelines.yml:14]:
```suggestion
    Linux_py39:
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -11,21 +11,27 @@ strategy:
...
+      vmImage: 'ubuntu-16.04'
+      python.version: '3.9'
     macOS_py37:
       vmImage: 'macOS-10.15'
       python.version: '3.7'
     macOS_py38:
       vmImage: 'macOS-latest'
       python.version: '3.8'
+    macOS_py38:
```
Comment (timhoffm) [inline azure-pipelines.yml:23]:
```suggestion
    macOS_py39:
```

File: azure-pipelines.yml [hunk 3]
```diff
@@ -11,21 +11,27 @@ strategy:
...
     Windows_py37:
       vmImage: 'vs2017-win2016'
       python.version: '3.7'
     Windows_py38:
       vmImage: 'windows-latest'
       python.version: '3.8'
-    Windows_pyPre:
-      vmImage: 'vs2017-win2016'
-      python.version: 'Pre'
+    Windows_py38:
```
Comment (timhoffm) [inline azure-pipelines.yml:32]:
```suggestion
    Windows_py39:
```

**Comment by tacaswell** [comment]
I missed that  #19353 existed.

**Review by QuLogic — APPROVED** [review]

**Review by dstansby — APPROVED** [review]

**Pull request merged**